src/components/directory-content.jsx

```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
+    const isActiveElementUnchecked = useMemo(
+        () =>
+            activeElement &&
+            !checkedElement.find(
+                (children) => children.elementUuid === activeElement.elementUuid
+            ),
+        [activeElement, checkedElement]
+    );
 
-    const getSelectedChildren = () => {
-        let selectedChildren = [];
-        if (currentChildren?.length > 0) {
-            // Adds the previously selected elements
-            if (selectedUuids?.size) {
-                selectedChildren = currentChildren
-                    .filter(
-                        (child) =>
-                            selectedUuids.has(child.elementUuid) &&
-                            child.elementUuid !== activeElement?.elementUuid
-                    )
-                    .map((child) => {
-                        return {
-                            subtype:
-                                childrenMetadata[child.elementUuid]?.subtype,
-                            hasMetadata:
-                                childrenMetadata[child.elementUuid] !==
-                                undefined,
-                            ...child,
-                        };
-                    });
-            }
+    const handleRowSelected = useCallback(() => {
+        setCheckedElement(computeCheckedElements(gridRef, childrenMetadata));
+    }, [childrenMetadata]);
 
-            // Adds the active element
-            if (activeElement) {
-                selectedChildren.push({
-                    ...activeElement,
-                    subtype:
-                        childrenMetadata[activeElement.elementUuid]?.subtype,
-                    hasMetadata:
-                        childrenMetadata[activeElement.elementUuid] !==
-                        undefined,
-                });
-            }
+    //It includes checked rows and the row with its context menu open
+    const fullSelection = useMemo(() => {
+        const selection = [...checkedElement];
+        if (isActiveElementUnchecked) {
+            selection.push(formatMetadata(activeElement, childrenMetadata));
         }
-        return [...new Set(selectedChildren)];
-    };
-
-    const rows = useMemo(
-        () =>
-            currentChildren?.map((child) => ({
-                ...child,
-                type:
-                    childrenMetadata[child.elementUuid] &&
-                    getElementTypeTranslation(
-                        child.type,
-                        childrenMetadata[child.elementUuid].subtype,
-                        childrenMetadata[child.elementUuid].format
-                    ),
-                notClickable: child.type === ElementType.CASE,
-            })),
-        [childrenMetadata, currentChildren, getElementTypeTranslation]
-    );
+        return selection;
+    }, [
+        activeElement,
+        checkedElement,
+        childrenMetadata,
+        isActiveElementUnchecked,
+    ]);
```

Comment:
currently, right clicking an element will add it to the selection
I think it will simplify the code if we keep this behaviour